packages/taiko-client-rs/crates/driver/tests/shasta_event_sync.rs

```diff
@@ -73,35 +63,32 @@ async fn syncs_shasta_proposal_into_l2() -> Result<()> {
             inbox_address: env.inbox_address,
         },
         Duration::from_millis(50),
-        RpcUrl::parse(blob_server.base_url().as_str())?,
+        blob_url.clone(),
         None,
-        Some(RpcUrl::parse(blob_server.base_url().as_str())?),
+        Some(blob_url.clone()),
     );
     let driver = Driver::new(driver_config).await?;
     let driver_client = driver.rpc_client().clone();
 
     let blob_source = Arc::new(
-        BlobDataSource::new(
-            Some(blob_server.endpoint().clone()),
-            Some(blob_server.endpoint().clone()),
-            true,
-        )
-        .await?,
+        BlobDataSource::new(Some(blob_endpoint.clone()), Some(blob_endpoint.clone()), true).await?,
     );
     let pipeline =
         ShastaDerivationPipeline::new(driver_client.clone(), blob_source, U256::ZERO).await?;
 
     let l2_head_before = driver_client.l2_provider.get_block_number().await?;
 
     let applier: &(dyn PayloadApplier + Send + Sync) = &driver_client;
+    let proposal_payload = indexer.get_last_proposal().unwrap();
```

Comment:
before we waited on `wait_for_new_proposal` are you sure it's not needed anymore ?